nmt-python-advanced-deploy-nemo-nmt-model-on-riva.ipynb

```diff
@@ -0,0 +1,404 @@
+    "\n",
+    "# Directory where the .riva model is stored\n",
+    "MODEL_LOC = \"<add path to model location>\"\n",
+    "# Example:\n",
+    "# import os\n",
+    "# MODEL_LOC = os.getcwd() + \"/NMTFinetuning/model\"\n",
+    "\n",
+    "# Name of the .riva file\n",
+    "MODEL_NAME = \"<add model name>\"\n",
+    "# Example:\n",
+    "# MODEL_NAME = \"en_es_24x6.riva\"\n",
+    "\n",
+    "# Key that model is encrypted with, while exporting with TAO\n",
+    "KEY = \"<add encryption key used for trained model>\"\n",
+    "# Example:\n",
+    "# KEY = \"tlt_encode\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f368c4bb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the ServiceMaker docker\n",
+    "! docker pull $RIVA_SM_CONTAINER"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "41eeb4fc",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Call riva-build command from the Riva Service Maker container.\n",
+    "# Example: \n",
+    "! docker run --rm --gpus 0 -v $MODEL_LOC:/data $RIVA_SM_CONTAINER bash -c \\\n",
+    "        \"riva-build translation --name en_es \\\n",
+    "            /data/en_es_24x6.rmir:key /data/en_es_24x6.riva:key\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3738f1d7",
+   "metadata": {},
+   "source": [
+    "### Riva-deploy\n",
+    "\n",
+    "The deployment tool takes as input one or more Riva Model Intermediate Representation (RMIR) files and a target model repository directory. It creates an ensemble configuration specifying the pipeline for the execution and finally writes all those assets to the output model repository directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bc934079",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Call riva-deploy command from the Riva Service Maker container.\n",
+    "#! docker run --rm --gpus 0 -v $MODEL_LOC:/data $RIVA_SM_CONTAINER bash -c \\\n",
+    "#        \"riva-deploy -f <rmir_filename>:$KEY <riva_model_target_repository>\"\n",
+    "# Example: \n",
+    "! docker run --rm --gpus 0 -v $MODEL_LOC:/data $RIVA_SM_CONTAINER bash -c \\\n",
+    "        \"riva-deploy -f /data/en_es_24x6.rmir:key /data/models && chmod -R 777 /data\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dcf64b4d",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "## Start the Riva Server\n",
+    "After the model repository is generated, we are ready to start the Riva server. First, download the [Riva Quick Start resource](https://ngc.nvidia.com/resources/riem1phmzvud:riva:riva_quickstart) from NGC. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d04bb9c7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!ngc registry resource download-version \"nvidia/riva/riva_quickstart:2.10.0\""
```

Comment:
Is it possible to use some kind of macro for release version so that it gets auto updated?

Reply:
Fixed.